torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -69,13 +68,28 @@ ncclDataType_t getNcclDataType(at::ScalarType type) {
 }
 
 ncclRedOp_t getNcclReduceOp(const ReduceOp reduceOp, at::Tensor& input) {
-  if (reduceOp == ReduceOp::SUM && input.scalar_type() == at::kBool) {
-    // For bool tensors, map sum to max, which both represent a bitwise or.
-    // This is to prevent overflow issues with sum, since we use uint8 to
-    // represent a bool (see ncclDataType mapping).
-    return ncclMax;
+  switch (reduceOp) {
+    case ReduceOp::BAND:
```

Comment:
We might add additional ops in the future (though I guess that is rare), would it be cleaner to just check if it's not present in the map and then throw?

Reply:
I am wondering if it's worth it since I need to know which op is actually being used to be able to throw a meaningful error. 🤔